tests/run/fused_bound_functions.pyx

```diff
@@ -0,0 +1,119 @@
+# mode: run
+# cython: binding=True
+
+"""
+Test that fused functions can be used in the same way as CyFunctions with respect to
+assigning them to class attributes. Previously they enforced extra type/argument checks
```

Comment:
Could you please also add the equivalent tests for Python functions and methods, by implementing the same thing in the doctests again? That way, we can be sure that all three function types really behave the same, not just the Cython implemented ones.

Alternatively, you can also change the test into a `.py` file and use pure Python declarations for the whole thing, so that the test runner will test it with both Python and Cython. That usually reduces the risk of incomplete test additions later.

Reply:
I've added equivalent tests to show that this is doing the same as Python.

Unfortunately writing it in pure Python mode doesn't seem possible at the minute https://github.com/cython/cython/issues/3142 (I get the same error using `@cython.locals` or annotations to set a fused type)

Reply:
Did you also try declaring the types with `cython.fused_type()`? Does that also not work here? There are tests for both in `test/run/pure_fused.py`.

Reply:
This looks like something that I fixed fairly recently https://github.com/cython/cython/pull/3391 but the branch I'm using for this PR is now a bit old and doesn't have that merged in. I'll have another go.

I suspect that accidently closes the linked issue too (not completely sure though)

Reply:
With master merged in I get a bit further, but I'm still having issues getting a pure version to work: https://github.com/cython/cython/issues/3460

Reply:
With all the other PRs merged and issues fixed now, I think it would be nice if this could become a pure Python test. It's really almost there, just a few declaration changes, but it makes a big difference in terms of compatibility testing.

Reply:
Done.